test/configuration_test.dart

```diff
@@ -153,4 +153,186 @@ Future<void> main([List<String>? args]) async {
     realm1.close();
     realm2.close();
   });
+
+  test('Configuration.initialDataCallback invoked', () {
+    var invoked = false;
+    var config = Configuration([Dog.schema, Person.schema]);
+    config.initialDataCallback = (realm) {
+      invoked = true;
+      realm.add(Dog('fido', owner: Person('john')));
+    };
+
+    final realm = Realm(config);
+    expect(invoked, true);
```

Comment:
Here you depend on core raising the callback immediately on the same thread that did the call. I expect this is an implementation detail, that core might change later.

Reply:
It's not - the event is raised just before the Realm is returned to the SDK and it's even the same Realm instance that will be returned to SDK. It can't be raised on another thread because we must block returning a Realm instance until the initial data callback has completed.

Reply:
Ties in with my misunderstanding of the API